SDL2/3 fails to build with SDL_WAYLAND_LIBDECOR_SHARED=OFF

SDL fails to build with SDL_WAYLAND_LIBDECOR_SHARED=OFF.

Steps to reproduce:

1. `cmake -DSDL_WAYLAND_LIBDECOR_SHARED=OFF -B build`
2. `cmake --build build`

This is with the latest SDL2 release and with the latest SDL3 commit as of today.

```
/home/user/projects/SDL/src/video/wayland/SDL_waylanddyn.c:94:45: error: unknown type name ‘SDL_DYNWAYLANDFN_libdecor_frame_get_min_content_size’; did you mean ‘SDL_DYNWAYLANDFN_libdecor_frame_set_min_content_size’?
   94 | #define SDL_WAYLAND_SYM_OPT(rc, fn, params) SDL_DYNWAYLANDFN_##fn WAYLAND_##fn = NULL;
      |                                             ^~~~~~~~~~~~~~~~~
/home/user/projects/SDL/src/video/wayland/SDL_waylandsym.h:224:1: note: in expansion of macro ‘SDL_WAYLAND_SYM_OPT’
  224 | SDL_WAYLAND_SYM_OPT(void, libdecor_frame_get_min_content_size, (struct libdecor_frame *,\
      | ^~~~~~~~~~~~~~~~~~~
In file included from /home/user/projects/SDL/src/video/wayland/SDL_waylanddyn.c:96:
/home/user/projects/SDL/src/video/wayland/SDL_waylandsym.h:224:1: warning: initialization of ‘int’ from ‘void *’ makes integer from pointer without a cast [-Wint-conversion]
/home/user/projects/SDL/src/video/wayland/SDL_waylanddyn.c:94:45: error: unknown type name ‘SDL_DYNWAYLANDFN_libdecor_frame_get_max_content_size’; did you mean ‘SDL_DYNWAYLANDFN_libdecor_frame_set_max_content_size’?
   94 | #define SDL_WAYLAND_SYM_OPT(rc, fn, params) SDL_DYNWAYLANDFN_##fn WAYLAND_##fn = NULL;
      |                                             ^~~~~~~~~~~~~~~~~
/home/user/projects/SDL/src/video/wayland/SDL_waylandsym.h:227:1: note: in expansion of macro ‘SDL_WAYLAND_SYM_OPT’
  227 | SDL_WAYLAND_SYM_OPT(void, libdecor_frame_get_max_content_size, (struct libdecor_frame *,\
      | ^~~~~~~~~~~~~~~~~~~
/home/user/projects/SDL/src/video/wayland/SDL_waylandsym.h:227:1: warning: initialization of ‘int’ from ‘void *’ makes integer from pointer without a cast [-Wint-conversion]
/home/user/projects/SDL/src/video/wayland/SDL_waylandsym.h: In function ‘SDL_WAYLAND_UnloadSymbols’:
/home/user/projects/SDL/src/video/wayland/SDL_waylanddyn.c:112:58: warning: assignment to ‘int’ from ‘void *’ makes integer from pointer without a cast [-Wint-conversion]
  112 | #define SDL_WAYLAND_SYM_OPT(rc, fn, params) WAYLAND_##fn = NULL;
      |                                                          ^
/home/user/projects/SDL/src/video/wayland/SDL_waylandsym.h:224:1: note: in expansion of macro ‘SDL_WAYLAND_SYM_OPT’
  224 | SDL_WAYLAND_SYM_OPT(void, libdecor_frame_get_min_content_size, (struct libdecor_frame *,\
      | ^~~~~~~~~~~~~~~~~~~
/home/user/projects/SDL/src/video/wayland/SDL_waylanddyn.c:112:58: warning: assignment to ‘int’ from ‘void *’ makes integer from pointer without a cast [-Wint-conversion]
  112 | #define SDL_WAYLAND_SYM_OPT(rc, fn, params) WAYLAND_##fn = NULL;
      |                                                          ^
/home/user/projects/SDL/src/video/wayland/SDL_waylandsym.h:227:1: note: in expansion of macro ‘SDL_WAYLAND_SYM_OPT’
  227 | SDL_WAYLAND_SYM_OPT(void, libdecor_frame_get_max_content_size, (struct libdecor_frame *,\
      | ^~~~~~~~~~~~~~~~~~~
/home/user/projects/SDL/src/video/wayland/SDL_waylandsym.h: In function ‘SDL_WAYLAND_LoadSymbols’:
/home/user/projects/SDL/src/video/wayland/SDL_waylanddyn.c:149:61: error: ‘SDL_DYNWAYLANDFN_libdecor_frame_get_min_content_size’ undeclared (first use in this function); did you mean ‘SDL_DYNWAYLANDFN_libdecor_frame_set_min_content_size’?
  149 | #define SDL_WAYLAND_SYM_OPT(rc, fn, params) WAYLAND_##fn = (SDL_DYNWAYLANDFN_##fn)WAYLAND_GetSym(#fn, thismod, SDL_FALSE);
      |                                                             ^~~~~~~~~~~~~~~~~
/home/user/projects/SDL/src/video/wayland/SDL_waylandsym.h:224:1: note: in expansion of macro ‘SDL_WAYLAND_SYM_OPT’
  224 | SDL_WAYLAND_SYM_OPT(void, libdecor_frame_get_min_content_size, (struct libdecor_frame *,\
      | ^~~~~~~~~~~~~~~~~~~
/home/user/projects/SDL/src/video/wayland/SDL_waylanddyn.c:149:61: note: each undeclared identifier is reported only once for each function it appears in
  149 | #define SDL_WAYLAND_SYM_OPT(rc, fn, params) WAYLAND_##fn = (SDL_DYNWAYLANDFN_##fn)WAYLAND_GetSym(#fn, thismod, SDL_FALSE);
      |                                                             ^~~~~~~~~~~~~~~~~
/home/user/projects/SDL/src/video/wayland/SDL_waylandsym.h:224:1: note: in expansion of macro ‘SDL_WAYLAND_SYM_OPT’
  224 | SDL_WAYLAND_SYM_OPT(void, libdecor_frame_get_min_content_size, (struct libdecor_frame *,\
      | ^~~~~~~~~~~~~~~~~~~
/home/user/projects/SDL/src/video/wayland/SDL_waylanddyn.c:149:83: error: expected ‘;’ before ‘WAYLAND_GetSym’
  149 | #define SDL_WAYLAND_SYM_OPT(rc, fn, params) WAYLAND_##fn = (SDL_DYNWAYLANDFN_##fn)WAYLAND_GetSym(#fn, thismod, SDL_FALSE);
      |                                                                                   ^~~~~~~~~~~~~~
/home/user/projects/SDL/src/video/wayland/SDL_waylandsym.h:224:1: note: in expansion of macro ‘SDL_WAYLAND_SYM_OPT’
  224 | SDL_WAYLAND_SYM_OPT(void, libdecor_frame_get_min_content_size, (struct libdecor_frame *,\
      | ^~~~~~~~~~~~~~~~~~~
/home/user/projects/SDL/src/video/wayland/SDL_waylanddyn.c:149:61: error: ‘SDL_DYNWAYLANDFN_libdecor_frame_get_max_content_size’ undeclared (first use in this function); did you mean ‘SDL_DYNWAYLANDFN_libdecor_frame_set_max_content_size’?
  149 | #define SDL_WAYLAND_SYM_OPT(rc, fn, params) WAYLAND_##fn = (SDL_DYNWAYLANDFN_##fn)WAYLAND_GetSym(#fn, thismod, SDL_FALSE);
      |                                                             ^~~~~~~~~~~~~~~~~
/home/user/projects/SDL/src/video/wayland/SDL_waylandsym.h:227:1: note: in expansion of macro ‘SDL_WAYLAND_SYM_OPT’
  227 | SDL_WAYLAND_SYM_OPT(void, libdecor_frame_get_max_content_size, (struct libdecor_frame *,\
      | ^~~~~~~~~~~~~~~~~~~
/home/user/projects/SDL/src/video/wayland/SDL_waylanddyn.c:149:83: error: expected ‘;’ before ‘WAYLAND_GetSym’
  149 | #define SDL_WAYLAND_SYM_OPT(rc, fn, params) WAYLAND_##fn = (SDL_DYNWAYLANDFN_##fn)WAYLAND_GetSym(#fn, thismod, SDL_FALSE);
      |                                                                                   ^~~~~~~~~~~~~~
/home/user/projects/SDL/src/video/wayland/SDL_waylandsym.h:227:1: note: in expansion of macro ‘SDL_WAYLAND_SYM_OPT’
  227 | SDL_WAYLAND_SYM_OPT(void, libdecor_frame_get_max_content_size, (struct libdecor_frame *,\
      | ^~~~~~~~~~~~~~~~~~~
```

--- a/src/video/wayland/SDL_waylanddyn.c
+++ b/src/video/wayland/SDL_waylanddyn.c
@@ -34,34 +34,29 @@ typedef struct
     const char *libname;
 } waylanddynlib;
 
-#ifndef SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_EGL
-#define SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_EGL NULL
-#endif
-#ifndef SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_CURSOR
-#define SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_CURSOR NULL
-#endif
-#ifndef SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_XKBCOMMON
-#define SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_XKBCOMMON NULL
-#endif
-#ifndef SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_LIBDECOR
-#define SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_LIBDECOR NULL
-#endif
-
 static waylanddynlib waylandlibs[] = {
+#ifdef SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_EGL
     { NULL, SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC },
-    { NULL, SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_EGL },
+#endif
+#ifdef SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_CURSOR
     { NULL, SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_CURSOR },
+#endif
+#ifdef SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_XKBCOMMON
     { NULL, SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_XKBCOMMON },
-    { NULL, SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_LIBDECOR }
+#endif
+#ifdef SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_LIBDECOR
+    { NULL, SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_LIBDECOR },
+#endif
+    { NULL, NULL }
 };
 
 static void *WAYLAND_GetSym(const char *fnname, int *pHasModule, SDL_bool required)
 {
-    int i;
     void *fn = NULL;
-    for (i = 0; i < SDL_TABLESIZE(waylandlibs); i++) {
-        if (waylandlibs[i].lib != NULL) {
-            fn = SDL_LoadFunction(waylandlibs[i].lib, fnname);
+    waylanddynlib *dynlib;
+    for (dynlib = waylandlibs; dynlib->libname; dynlib++) {
+        if (dynlib->lib != NULL) {
+            fn = SDL_LoadFunction(dynlib->lib, fnname);
             if (fn != NULL) {
                 break;
             }
@@ -69,10 +64,11 @@ static void *WAYLAND_GetSym(const char *fnname, int *pHasModule, SDL_bool requir
     }
 
 #if DEBUG_DYNAMIC_WAYLAND
-    if (fn != NULL)
-        SDL_Log("WAYLAND: Found '%s' in %s (%p)\n", fnname, waylandlibs[i].libname, fn);
-    else
+    if (fn != NULL) {
+        SDL_Log("WAYLAND: Found '%s' in %s (%p)\n", fnname, dynlib->libname, fn);
+    } else {
         SDL_Log("WAYLAND: Symbol '%s' NOT FOUND!\n", fnname);
+    }
 #endif
 
     if (fn == NULL && required) {

